src/AzureIoTHub.Portal.Server/Jobs/SyncDeviceTelemetryJob.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) CGI France. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace AzureIoTHub.Portal.Server.Jobs
+{
+    using System;
+    using System.Linq;
+    using System.Text.Json;
+    using System.Threading.Tasks;
+    using Azure.Messaging.EventHubs.Consumer;
+    using Domain;
+    using Domain.Entities;
+    using Domain.Repositories;
+    using Microsoft.Extensions.Logging;
+    using Quartz;
+    using System.Globalization;
+
+    public class SyncDeviceTelemetryJob : IJob
+    {
+        private readonly ILogger<SyncDeviceTelemetryJob> logger;
+        private readonly ConfigHandler configHandler;
+        private readonly ILorawanDeviceRepository lorawanDeviceRepository;
+        private readonly IDeviceTelemetryRepository deviceTelemetryRepository;
+        private readonly IUnitOfWork unitOfWork;
+
+        public SyncDeviceTelemetryJob(ILogger<SyncDeviceTelemetryJob> logger,
+            ConfigHandler configHandler,
+            ILorawanDeviceRepository lorawanDeviceRepository,
+            IDeviceTelemetryRepository deviceTelemetryRepository,
+            IUnitOfWork unitOfWork)
+        {
+            this.logger = logger;
+            this.configHandler = configHandler;
+            this.lorawanDeviceRepository = lorawanDeviceRepository;
+            this.deviceTelemetryRepository = deviceTelemetryRepository;
+            this.unitOfWork = unitOfWork;
+        }
+
+        public async Task Execute(IJobExecutionContext context)
+        {
+            await using var consumer = new EventHubConsumerClient(this.configHandler.IoTHubEventHubConsumerGroup, this.configHandler.IoTHubEventHubEndpoint);
```

Comment:
It will be better here to use ``EventProcessorClient`` instead of ``EventHubConsumerClient``. This will give us the possibility to handle production scenarios: 
- Multiple instances reading events
- Checkpointing the offsets to Azure Storage
- ...